### ISSUE TITLE
5.1 Changing default Markdown extension to .wiki

### ISSUE THREAD

**Issue by olablt** [gollum/gollum#1561]
Hello all,

I followed instructions found here
https://davidyat.es/2017/09/01/vimwiki-plus-gollum/

Created config.rb
```ruby
# always include this:
Gollum::Page.send :remove_const, :FORMAT_NAMES if defined? Gollum::Page::FORMAT_NAMES

# first remove all markups:
Gollum::Markup.formats.clear

# and then define the sole markup to be supported:
Gollum::Markup.formats[:markdown] = {
    :name => "Markdown",
    :extensions => ['wiki', 'md'],
    :regexp =>/wiki|md|mkdn?|mdown|markdown/ ,
    :reverse_links => false
}
```

then started
```sh
$ mkdir tmp && cd tmp
$ git init
$ gollum --version
Gollum 5.0.1
$ gollum --versions
/usr/local/share/gems/gems/gollum-lib-5.0.4/lib/gollum-lib/markups.rb:25: warning: Insecure world writable dir /home/bob/Projects in PATH, mode 040766
Gollum 5.0.1
Running on: x86_64-linux with Ruby version 2.6.5
Using:
rugged 0.99.0
gollum-rugged_adapter 1.0
gollum-lib 5.0.4
Markdown rendering gem: kramdown
Other renderering gems:
none
$ gollum --config=config.rb
```

but Markdown files are created with "md", not "wiki"

**Comment by olablt** [comment]
false alarm. Found instructions here
https://github.com/gollum/gollum/wiki/Formats-and-extensions

and updated my config.rb to
```ruby
# always include this:
Gollum::Page.send :remove_const, :FORMAT_NAMES if defined? Gollum::Page::FORMAT_NAMES

# first remove all markups:
Gollum::Markup.formats.clear

# and then define the sole markup to be supported:
Gollum::Markup.formats[:wiki] = {
    :name => "Markdown",
    :extensions => ['wiki', 'md'],
    :regexp =>/wiki|md|mkdn?|mdown|markdown/ ,
    # :reverse_links => false
}
```

**Comment by olablt** [comment]
markdown rendering not working..

```
# heading1

paragraph1

paragraph2

## heading2

```

results to

![image](https://user-images.githubusercontent.com/674574/80921884-f2154800-8d81-11ea-8d1e-a219d0cf38fc.png)

**Comment by olablt** [comment]
@dmyates could you help on this? should I change my gollum version?..

**Comment by dmyates** [comment]
Hey man, what version of Gollum are you using? If you're on 5.x, try changing

```ruby
Gollum::Markup.formats[:wiki]
```
to
```ruby
Gollum::Markup.formats[:markdown]
```

**Comment by dometto** [comment]
I think your markup definition should simply be the following:

```ruby
Gollum::Markup.formats[:markdown] = {
    :name => "Markdown",
    :extensions => ['wiki', 'md']
}
```

`:regexp` deprecated. If this works out for you, please update the [wiki](https://github.com/gollum/gollum/wiki/Formats-and-extensions).

**Comment by dometto** [comment]
This was a little more complicated than I thought, and requires a [fix](https://github.com/gollum/gollum-lib/pull/378) to gollum-lib to be able to use a custom extension as the default new filename for a given format. Once that fix is released, the following snippet works:

```
# Copy the old format definition
markdown = Gollum::Markup.formats[:markdown]
# Remove the old format definition
Gollum::Markup.formats.delete(:markdown)
# Modify the format definition
markdown[:extensions] = ['wiki', 'md'] # The first extension will be used as the default extension for this format (when creating a new page)
# Re-register the format
Gollum::Markup.register(:markdown, 'Markdown', markdown)
```

I've updated the [wiki](https://github.com/gollum/gollum/wiki/Formats-and-extensions) accordingly.

**Comment by dometto** [comment]
Fix released in gollum-lib 5.0.5
